src/Middleware/RateLimiting/src/RateLimitingMiddleware.cs

```diff
@@ -139,21 +145,30 @@ private LeaseContext CombinedAcquire(HttpContext context)
                 globalLease = _globalLimiter.AttemptAcquire(context);
                 if (!globalLease.IsAcquired)
                 {
-                    return new LeaseContext() { GlobalRejected = true, Lease = globalLease };
+                    return new LeaseContext() { RequestRejectionReason = RequestRejectionReason.GlobalLimiter, Lease = globalLease };
                 }
             }
             endpointLease = _endpointLimiter.AttemptAcquire(context);
             if (!endpointLease.IsAcquired)
             {
                 globalLease?.Dispose();
-                return new LeaseContext() { GlobalRejected = false, Lease = endpointLease };
+                return new LeaseContext() { RequestRejectionReason = RequestRejectionReason.EndpointLimiter, Lease = endpointLease };
             }
         }
-        catch (Exception)
+        catch (Exception ex)
         {
             endpointLease?.Dispose();
             globalLease?.Dispose();
-            throw;
+            // Don't throw if the request was canceled - instead log. 
+            if (ex is OperationCanceledException && context.RequestAborted.IsCancellationRequested)
```

Comment:
This code path won't throw on request aborted will it?

Reply:
It should log debug then return from the middleware (line 90)

Reply:
I mean, you call AttemptAcquire which doesn't take a cancellation token, so who would be throwing the OCE?

Reply:
Oh I see what you're saying - yeah, you're right, this was an unnecessary copy/paste. I'll remove.